### PR TITLE
explorer: display package module execute tab

### PR DESCRIPTION
**Pull request by pchrysochoidis**
* adds feature flag - on only in dev
* adds execute tab in package module view
* displays each entry function in a collapsible box with function details as json

https://user-images.githubusercontent.com/10210143/203170415-fb2002e9-72d5-4722-a1e1-964658ce2eca.mov

part of APPS-41

### CODE REVIEW

**Review by mystie711 — APPROVED** [review]

File: apps/explorer/src/components/module/module-functions-interaction/ModuleFunction.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    functionDetails: SuiMoveNormalizedFunction;
+    defaultOpen?: boolean;
+};
+export function ModuleFunction({
+    defaultOpen,
+    functionName,
+    functionDetails,
+}: ModuleFunctionProps) {
+    return (
+        <div className="rounded-lg bg-sui-grey-40">
```
Comment (Jordan-Mysten) [inline apps/explorer/src/components/module/module-functions-interaction/ModuleFunction.tsx:24]: Seems like a good thing to pull out into a base UI component

File: apps/explorer/src/components/module/module-functions-interaction/ModuleFunction.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            <Disclosure defaultOpen={defaultOpen}>
+                <Disclosure.Button
+                    as="div"
+                    className="flex flex-nowrap cursor-pointer items-center pt-[15px] pb-[17px] px-[20px]"
```
Comment (Jordan-Mysten) [inline apps/explorer/src/components/module/module-functions-interaction/ModuleFunction.tsx:28]: Why all the custom padding values?

File: apps/explorer/src/components/module/module-functions-interaction/ModuleFunction.tsx (hunk 1 above)
Comment (pchrysochoidis) [reply apps/explorer/src/components/module/module-functions-interaction/ModuleFunction.tsx:24]: Good catch I will move it there

File: apps/explorer/src/components/module/module-functions-interaction/ModuleFunction.tsx (hunk 2 above)
Comment (pchrysochoidis) [reply apps/explorer/src/components/module/module-functions-interaction/ModuleFunction.tsx:28]: Actually 20 shouldn't be custom, but for the rest trying to follow the design. I guess I could use `pt-3.5` and `pb-4` 1px should be ok, @mystie711 ?

File: apps/explorer/src/components/module/module-functions-interaction/ModuleFunction.tsx (hunk 1 above)
Comment (pchrysochoidis) [reply apps/explorer/src/components/module/module-functions-interaction/ModuleFunction.tsx]: done

File: apps/explorer/src/components/module/module-functions-interaction/ModuleFunction.tsx (hunk 2 above)
Comment (pchrysochoidis) [reply apps/explorer/src/components/module/module-functions-interaction/ModuleFunction.tsx]: done

File: apps/explorer/src/utils/growthbook.ts [hunk 3]
```diff
@@ -29,4 +29,5 @@ export async function loadFeatures() {
 
 export enum GROWTHBOOK_FEATURES {
     USE_TEST_NET_ENDPOINT = 'testnet-selection',
+    MODULE_VIEW_INVOKE_FUNCTIONS = 'explorer-module-view-invoke-functions',
```
Comment (Jordan-Mysten) [inline apps/explorer/src/utils/growthbook.ts:32]: I love that we feature flagged this, makes it so much easier to land this incrementally!

File: apps/explorer/src/ui/DisclosureBox.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    defaultOpen,
+    title,
+    children,
+}: DisclosureBoxProps) {
+    return (
+        <div className="rounded-lg bg-sui-grey-40">
+            <Disclosure defaultOpen={defaultOpen}>
+                <Disclosure.Button
+                    as="div"
+                    className="flex flex-nowrap cursor-pointer items-center pt-3.5 pb-4 px-5"
```
Comment (Jordan-Mysten) [inline apps/explorer/src/ui/DisclosureBox.tsx:26]: I don't understand why we have variable top vs bottom padding, seems off.

File: apps/explorer/src/ui/DisclosureBox.tsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { Disclosure } from '@headlessui/react';
+
+import type { ReactNode } from 'react';
+
+import { ReactComponent as ChevronDownIcon } from '~/assets/SVGIcons/16px/chevron-down.svg';
```
Comment (Jordan-Mysten) [inline apps/explorer/src/ui/DisclosureBox.tsx:8]: Move this to `icons` in `ui` to align with the rest of the new components.

File: apps/explorer/src/ui/DisclosureBox.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+export type DisclosureBoxProps = {
+    defaultOpen?: boolean;
+    title: ReactNode;
+    children: ReactNode | ReactNode[];
```
Comment (Jordan-Mysten) [inline apps/explorer/src/ui/DisclosureBox.tsx:13]: Should just be `ReactNode`

File: apps/explorer/src/ui/DisclosureBox.tsx [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+                <Disclosure.Button
+                    as="div"
+                    className="flex flex-nowrap cursor-pointer items-center pt-3.5 pb-4 px-5"
+                >
+                    <div className="flex-1 text-body font-semibold text-sui-grey-90">
+                        {title}
+                    </div>
+                    <ChevronDownIcon className="-rotate-90 ui-open:rotate-0 text-sui-grey-75" />
+                </Disclosure.Button>
+                <Disclosure.Panel className="text-gray-500 px-5 pb-5">
```
Comment (Jordan-Mysten) [inline apps/explorer/src/ui/DisclosureBox.tsx:33]: Probably remove the text color.

File: apps/explorer/src/hooks/useNormalizedMoveModule.ts [hunk 8]
```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+import { useQuery } from '@tanstack/react-query';
+
+import { useRpc } from './useRpc';
+
+import type { ObjectId } from '@mysten/sui.js';
+
+export function useNormalizedMoveModule(
+    packageId: ObjectId | null | undefined,
```
Comment (Jordan-Mysten) [inline apps/explorer/src/hooks/useNormalizedMoveModule.ts:10]: nit but I prefer expressing these as `packageId?: ObjectId | null`,

File: apps/explorer/src/components/module/module-functions-interaction/index.tsx [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+        );
+    }
+
+    return !isLoading && executableFunctions.length ? (
+        <div className="flex flex-col gap-3">
+            {executableFunctions.map(({ name, details }, index) => (
+                <ModuleFunction
+                    key={name}
+                    functionName={name}
+                    defaultOpen={index === 0}
```
Comment (Jordan-Mysten) [inline apps/explorer/src/components/module/module-functions-interaction/index.tsx:50]: I feel like we can just have these all be false and force the user to expand any of them

**Comment by pchrysochoidis** [comment]
Thanks @Jordan-Mysten, I think I addressed all the comments. Let me know if there is anything else!

File: apps/core/tailwind.config.js [hunk 10]
```diff
@@ -121,6 +121,7 @@ module.exports = {
         shimmer: "shimmer 2s infinite linear",
       },
       spacing: {
+        3.75: "0.9375rem",
```
Comment (Jordan-Mysten) [inline apps/core/tailwind.config.js:124]: I really wish we had a more consistent spacing scale (cc @mystie711)

**Review by Jordan-Mysten — APPROVED** [review]

**Pull request merged**